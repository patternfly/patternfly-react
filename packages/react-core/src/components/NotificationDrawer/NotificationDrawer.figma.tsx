import React from 'react';
import { NotificationDrawer } from './NotificationDrawer';
import figma from '@figma/code-connect';

/**
 * -- This file was auto-generated by Code Connect --
 * `props` includes a mapping from Figma properties and variants to
 * suggested values. You should update this to match the props of your
 * code component, and update the `example` function to return the
 * code example you'd like to see in Figma
 */

figma.connect(
  NotificationDrawer,
  'https://www.figma.com/design/VMEX8Xg2nzhBX8rfBx53jp/branch/H3LonYnwH26v9zNEa2SXFk/PatternFly-6%3A-Components?m=auto&node-id=7172-99015',
  {
    props: {
      type: figma.enum('Type', {
        Basic: 'basic',
        Grouped: 'grouped'
      })
    },
    example: (props) => <NotificationDrawer />
  }
);

figma.connect(
  NotificationDrawer,
  'https://www.figma.com/design/VMEX8Xg2nzhBX8rfBx53jp/branch/H3LonYnwH26v9zNEa2SXFk/PatternFly-6%3A-Components?m=auto&node-id=3172-18190',
  {
    props: {
      hasCount: figma.boolean('Has count'),
      groupTitle: figma.string('Group title'),
      type: figma.enum('Type', {
        Collapsed: 'collapsed',
        Expanded: 'expanded'
      })
    },
    example: (props) => <NotificationDrawer />
  }
);

figma.connect(
  NotificationDrawer,
  'https://www.figma.com/design/VMEX8Xg2nzhBX8rfBx53jp/branch/H3LonYnwH26v9zNEa2SXFk/PatternFly-6%3A-Components?m=auto&node-id=3170-17841',
  {
    props: {
      showUnreadCount: figma.boolean('Show unread count'),
      unreadCount: figma.string('Unread count'),
      headingText: figma.string('Heading text'),
      hasActionsMenu: figma.boolean('Has actions menu')
    },
    example: (props) => <NotificationDrawer />
  }
);

figma.connect(
  NotificationDrawer,
  'https://www.figma.com/design/VMEX8Xg2nzhBX8rfBx53jp/branch/H3LonYnwH26v9zNEa2SXFk/PatternFly-6%3A-Components?m=auto&node-id=3164-16861',
  {
    props: {
      status: figma.enum('Status', {
        Danger: 'danger',
        Info: 'info',
        Success: 'success',
        Warning: 'warning'
      }),
      type: figma.enum('Type', {
        Unread: 'unread',
        Read: 'read'
      }),
      state: figma.enum('State', {
        Default: 'default',
        Hover: 'hover'
      }),
      topActionType: figma.enum('Top Action Type', {
        'Icon button': 'icon-button',
        'Link button': 'link-button'
      })
    },
    example: (props) => <NotificationDrawer />
  }
);
