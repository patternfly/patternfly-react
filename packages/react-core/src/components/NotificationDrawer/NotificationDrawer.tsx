import * as React from 'react';

import styles from '@patternfly/react-styles/css/components/NotificationDrawer/notification-drawer';
import { css } from '@patternfly/react-styles';

export interface NotificationDrawerProps extends React.HTMLProps<HTMLDivElement> {
  /**  Content rendered inside the notification drawer */
  children?: React.ReactNode;
  /**  Additional classes added to the notification drawer */
  className?: string;
  /** @hide Forwarded ref */
  innerRef?: React.Ref<any>;
}

const NotificationDrawerBase: React.FunctionComponent<NotificationDrawerProps> = ({
  children,
  className = '',
  innerRef,
  ...props
}: NotificationDrawerProps) => (
  <div ref={innerRef} {...props} className={css(styles.notificationDrawer, className)}>
    {children}
  </div>
);
export const NotificationDrawer = React.forwardRef((props: NotificationDrawerProps, ref: React.Ref<any>) => (
  <NotificationDrawerBase innerRef={ref} {...props} />
));
NotificationDrawer.displayName = 'NotificationDrawer';
