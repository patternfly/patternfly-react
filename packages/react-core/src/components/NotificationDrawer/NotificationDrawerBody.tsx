import * as React from 'react';

import styles from '@patternfly/react-styles/css/components/NotificationDrawer/notification-drawer';
import { css } from '@patternfly/react-styles';

export interface NotificationDrawerBodyProps extends React.HTMLProps<HTMLDivElement> {
  /**  Content rendered inside the body of the notification drawer */
  children?: React.ReactNode;
  /**  Additional classes added to the notification drawer body */
  className?: string;
}

export const NotificationDrawerBody: React.FunctionComponent<NotificationDrawerBodyProps> = ({
  children,
  className = '',
  ...props
}: NotificationDrawerBodyProps) => (
  <div {...props} className={css(styles.notificationDrawerBody, className)}>
    {children}
  </div>
);
NotificationDrawerBody.displayName = 'NotificationDrawerBody';
