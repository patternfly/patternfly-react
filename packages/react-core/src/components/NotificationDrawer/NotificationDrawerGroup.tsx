import * as React from 'react';

import AngleRightIcon from '@patternfly/react-icons/dist/js/icons/angle-right-icon';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/NotificationDrawer/notification-drawer';
import maxLines from '@patternfly/react-tokens/dist/js/c_notification_drawer__group_toggle_title_max_lines';

import { Badge } from '../Badge';
import { Tooltip } from '../Tooltip';

export interface NotificationDrawerGroupProps extends React.HTMLProps<HTMLElement> {
  /**  Content rendered inside the group */
  children?: React.ReactNode;
  /**  Additional classes added to the group */
  className?: string;
  /**  Notification drawer group count */
  count: number;
  /**  Adds styling to the group to indicate expanded state */
  isExpanded: boolean;
  /**  Adds styling to the group to indicate whether it has been read */
  isRead?: boolean;
  /**  Callback for when group button is clicked to expand */
  onExpand?: (event: any, value: boolean) => void;
  /**  Notification drawer group title */
  title: string;
  /** Truncate title to number of lines */
  truncateTitle?: number;
  /** Position of the tooltip which is displayed if text is truncated */
  tooltipPosition?: 'auto' | 'top' | 'bottom' | 'left' | 'right';
}

export const NotificationDrawerGroup: React.FunctionComponent<NotificationDrawerGroupProps> = ({
  children,
  className = '',
  count,
  isExpanded,
  isRead = false,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  onExpand = (event: any, expanded: boolean) => undefined as any,
  title,
  truncateTitle = 0,
  tooltipPosition,
  ...props
}: NotificationDrawerGroupProps) => {
  const titleRef = React.useRef(null);
  const [isTooltipVisible, setIsTooltipVisible] = React.useState(false);
  React.useEffect(() => {
    // Title will always truncate on overflow regardless of truncateTitle prop
    const showTooltip = titleRef.current && titleRef.current.offsetHeight < titleRef.current.scrollHeight;
    if (isTooltipVisible !== showTooltip) {
      setIsTooltipVisible(showTooltip);
    }
    if (!titleRef.current || !truncateTitle) {
      return;
    }
    titleRef.current.style.setProperty(maxLines.name, truncateTitle.toString());
  }, [titleRef, truncateTitle, isTooltipVisible]);

  const Title = (
    <div
      {...(isTooltipVisible && { tabIndex: 0 })}
      ref={titleRef}
      className={css(styles.notificationDrawerGroupToggleTitle)}
    >
      {title}
    </div>
  );

  return (
    <section
      {...props}
      className={css(styles.notificationDrawerGroup, isExpanded && styles.modifiers.expanded, className)}
    >
      <h1>
        <button
          className={css(styles.notificationDrawerGroupToggle)}
          aria-expanded={isExpanded}
          onClick={e => onExpand(e, !isExpanded)}
          onKeyDown={e => {
            if (e.key === 'Enter' || e.key === ' ') {
              e.preventDefault();
              onExpand(e, !isExpanded);
            }
          }}
        >
          {isTooltipVisible ? (
            <Tooltip content={title} position={tooltipPosition}>
              {Title}
            </Tooltip>
          ) : (
            Title
          )}
          <div className={css(styles.notificationDrawerGroupToggleCount)}>
            <Badge isRead={isRead}>{count}</Badge>
          </div>

          <span className="pf-c-notification-drawer__group-toggle-icon">
            <AngleRightIcon />
          </span>
        </button>
      </h1>
      {children}
    </section>
  );
};
NotificationDrawerGroup.displayName = 'NotificationDrawerGroup';
