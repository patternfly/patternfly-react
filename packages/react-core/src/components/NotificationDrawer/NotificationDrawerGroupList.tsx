import * as React from 'react';

import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/NotificationDrawer/notification-drawer';

export interface NotificationDrawerGroupListProps extends React.HTMLProps<HTMLDivElement> {
  /**  Content rendered inside the notification drawer list body */
  children?: React.ReactNode;
  /**  Additional classes added to the notification drawer list body */
  className?: string;
}

export const NotificationDrawerGroupList: React.FunctionComponent<NotificationDrawerGroupListProps> = ({
  children,
  className = '',
  ...props
}: NotificationDrawerGroupListProps) => (
  <div {...props} className={css(styles.notificationDrawerGroupList, className)}>
    {children}
  </div>
);
NotificationDrawerGroupList.displayName = 'NotificationDrawerGroupList';
