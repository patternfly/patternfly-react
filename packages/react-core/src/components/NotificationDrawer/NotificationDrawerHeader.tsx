import * as React from 'react';

import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/NotificationDrawer/notification-drawer';
import TimesIcon from '@patternfly/react-icons/dist/js/icons/times-icon';

import { Text, TextVariants } from '../Text';
import { Button, ButtonVariant } from '../Button';

export interface NotificationDrawerHeaderProps extends React.HTMLProps<HTMLDivElement> {
  /**  Content rendered inside the drawer */
  children?: React.ReactNode;
  /**  Additional classes for notification drawer header. */
  className?: string;
  /** Adds custom accessible text to the notification drawer close button. */
  closeButtonAriaLabel?: string;
  /**  Notification drawer heading count */
  count?: number;
  /**  Notification drawer heading custom text which can be used instead of providing count/unreadText */
  customText?: string;
  /**  Callback for when close button is clicked */
  onClose?: () => void;
  /**  Notification drawer heading title */
  title?: string;
  /**  Notification drawer heading unread text used in combination with a count */
  unreadText?: string;
}

export const NotificationDrawerHeader: React.FunctionComponent<NotificationDrawerHeaderProps> = ({
  children,
  className = '',
  count,
  closeButtonAriaLabel = 'Close',
  customText,
  onClose,
  title = 'Notifications',
  unreadText = 'unread',
  ...props
}: NotificationDrawerHeaderProps) => (
  <div {...props} className={css(styles.notificationDrawerHeader, className)}>
    <Text component={TextVariants.h1} className={css(styles.notificationDrawerHeaderTitle)}>
      {title}
    </Text>
    {(customText !== undefined || count !== undefined) && (
      <span className={css(styles.notificationDrawerHeaderStatus)}>{customText || `${count} ${unreadText}`}</span>
    )}
    {children && (
      <div className={css(styles.notificationDrawerHeaderAction)}>
        {children}
        {onClose && (
          <div>
            <Button variant={ButtonVariant.plain} aria-label={closeButtonAriaLabel} onClick={onClose}>
              <TimesIcon aria-hidden="true" />
            </Button>
          </div>
        )}
      </div>
    )}
  </div>
);
NotificationDrawerHeader.displayName = 'NotificationDrawerHeader';
