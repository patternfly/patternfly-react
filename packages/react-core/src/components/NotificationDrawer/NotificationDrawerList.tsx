import * as React from 'react';

import { css } from '@patternfly/react-styles';

export interface NotificationDrawerListProps extends React.HTMLProps<HTMLUListElement> {
  /**  Content rendered inside the notification drawer list body */
  children?: React.ReactNode;
  /**  Additional classes added to the notification drawer list body */
  className?: string;
  /**  Adds styling to the notification drawer list to indicate expand/hide state */
  isHidden?: boolean;
}

export const NotificationDrawerList: React.FunctionComponent<NotificationDrawerListProps> = ({
  children,
  className = '',
  isHidden = false,
  ...props
}: NotificationDrawerListProps) => (
  <ul {...props} className={css('pf-c-notification-drawer__list', className)} hidden={isHidden}>
    {children}
  </ul>
);
NotificationDrawerList.displayName = 'NotificationDrawerList';
