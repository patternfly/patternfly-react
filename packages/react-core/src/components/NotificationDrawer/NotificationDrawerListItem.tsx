import * as React from 'react';
import { css } from '@patternfly/react-styles';
import styles from '@patternfly/react-styles/css/components/NotificationDrawer/notification-drawer';

export interface NotificationDrawerListItemProps extends React.HTMLProps<HTMLLIElement> {
  /**  Content rendered inside the list item */
  children?: React.ReactNode;
  /**  Additional classes added to the list item */
  className?: string;
  /**  Modifies the list item to include hover styles on :hover */
  isHoverable?: boolean;
  /**  Adds styling to the list item to indicate it has been read */
  isRead?: boolean;
  /**  Callback for when a list item is clicked */
  onClick?: (event: any) => void;
  /**  Tab index for the list item */
  tabIndex?: number;
  /**  Variant indicates the severity level */
  variant?: 'default' | 'success' | 'danger' | 'warning' | 'info';
}

export const NotificationDrawerListItem: React.FunctionComponent<NotificationDrawerListItemProps> = ({
  children = null,
  className = '',
  isHoverable = true,
  isRead = false,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  onClick = (event: React.MouseEvent) => undefined as any,
  tabIndex = 0,
  variant = 'default',
  ...props
}: NotificationDrawerListItemProps) => {
  const onKeyDown = (event: React.KeyboardEvent) => {
    // Accessibility function. Click on the list item when pressing Enter or Space on it.
    if (event.key === 'Enter' || event.key === ' ') {
      (event.target as HTMLElement).click();
    }
  };
  return (
    <li
      {...props}
      className={css(
        styles.notificationDrawerListItem,
        isHoverable && styles.modifiers.hoverable,
        styles.modifiers[variant],
        isRead && styles.modifiers.read,
        className
      )}
      tabIndex={tabIndex}
      onClick={e => onClick(e)}
      onKeyDown={onKeyDown}
    >
      {children}
    </li>
  );
};
NotificationDrawerListItem.displayName = 'NotificationDrawerListItem';
