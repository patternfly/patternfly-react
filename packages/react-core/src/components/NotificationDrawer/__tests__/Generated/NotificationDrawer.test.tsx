/**
 * This test was generated
 */
import * as React from 'react';
import { render } from '@testing-library/react';
import { NotificationDrawer } from '../../NotificationDrawer';
// any missing imports can usually be resolved by adding them here
import {} from '../..';

it('NotificationDrawer should match snapshot (auto-generated)', () => {
  const view = render(<NotificationDrawer children={<div>ReactNode</div>} className={"''"} />);
  expect(view.container).toMatchSnapshot();
});
