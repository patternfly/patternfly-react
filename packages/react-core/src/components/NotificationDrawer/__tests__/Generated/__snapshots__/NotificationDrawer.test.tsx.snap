// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationDrawer should match snapshot (auto-generated) 1`] = `
<div>
  <div
    class="pf-c-notification-drawer ''"
  >
    <div>
      ReactNode
    </div>
  </div>
</div>
`;
